torch/_guards.py

```diff
@@ -611,6 +611,8 @@ def __init__(self, fake_mode):
         # ints that are known to be size-like and may have 0/1 entries that we
         # must not specialize on.
         self.force_unspec_int_unbacked_size_like = False
+        # TODO(voz): Is this too silent? Should this be an assert that says "You must turn on compiled autograd"
+        self.requires_compiled_autograd = False
```

Comment:
I could totally see an argument for us making this a hard assert (you must have compiled autograd on) or a graph break (We cant do foo in forward, we arent in compiled autograd, bail safely), or always proceeding (as this PR does) and then switching off compiled_bwd in aot_autograd.

Do we have a preference?